tools/project_manager/update_illos.php

```diff
@@ -62,7 +62,9 @@
     exit;
 }
 
-if (($is_delete_all_operation || $operation == 'delete') && $project->state != PROJ_NEW && $project->state != PROJ_P1_UNAVAILABLE) {
+if (($is_delete_all_operation || $operation == 'delete') && !$project->user_can_delete_nonpage_images) {
```

Comment:
don't you have to invoke the function `!$project->user_can_delete_nonpage_images()`? My php is bad though so I may be wrong here.

Reply:
This is a PHP feature. The `Project` class uses the `__get()` [magic function](https://www.php.net/manual/en/language.oop5.overloading.php#object.set) that allows this to be accessed like a property rather than a function. Sharon wired this into that magic function at line 603 in `Project.inc`.

Reply:
Say, rather, that it seemed like I **_had_** to wire it into the magic function. 😁  -- I was following the pattern of what had been done before, and that broke things without fixing that, too. (And then, too, even if there's another way to do things, it's better to follow the current pattern if possible, unless there's a reason to rip it all out and redo it.)

Reply:
Oh I agree, I just wasn't aware of this __get thing. Thanks for explaining with links @cpeel